exercises/react/test/test_react.c

```diff
@@ -0,0 +1,289 @@
+#include <stddef.h>
+#include <stdio.h>
+#include "vendor/unity.h"
+#include "../src/react.h"
+
+void test_input_cells_have_value(void)
+{
+   struct reactor *r = create_reactor();
+   struct cell *input = create_input_cell(r, 2);
+
+   TEST_ASSERT_EQUAL_INT(2, cell_value(input));
+
+   free_reactor(r);
+}
+
+static int plus1(int x)
+{
+   return x + 1;
+}
+
+void test_compute_cells_calculate_initial_value(void)
+{
+   struct reactor *r = create_reactor();
+   struct cell *input = create_input_cell(r, 1);
+   struct cell *output = create_compute1_cell(r, input, plus1);
+
+   TEST_ASSERT_EQUAL_INT(2, cell_value(output));
+
+   free_reactor(r);
+}
+
+static int concat_digits(int a, int b)
+{
+   return b * 10 + a;
+}
+
+void test_compute_cells_take_inputs_in_the_right_order(void)
+{
+   struct reactor *r = create_reactor();
+   struct cell *one = create_input_cell(r, 1);
+   struct cell *two = create_input_cell(r, 2);
+   struct cell *output = create_compute2_cell(r, one, two, concat_digits);
+
+   TEST_ASSERT_EQUAL_INT(21, cell_value(output));
+
+   free_reactor(r);
+}
+
+void test_compute_cells_update_value_when_dependencies_are_changed(void)
+{
+   struct reactor *r = create_reactor();
+   struct cell *input = create_input_cell(r, 1);
+   struct cell *output = create_compute1_cell(r, input, plus1);
+
+   cell_value_set(input, 3);
+   TEST_ASSERT_EQUAL_INT(4, cell_value(output));
+
+   free_reactor(r);
+}
+
+static int times2(int x)
+{
+   return x * 2;
+}
+
+static int times30(int x)
+{
+   return x * 30;
+}
+
+static int plus(int x, int y)
+{
+   return x + y;
+}
+
+void test_compute_cells_can_depend_on_other_compute_cells(void)
+{
+   struct reactor *r = create_reactor();
+   struct cell *input = create_input_cell(r, 1);
+   struct cell *times_two = create_compute1_cell(r, input, times2);
+   struct cell *times_thirty = create_compute1_cell(r, input, times30);
+   struct cell *output = create_compute2_cell(r, times_two, times_thirty, plus);
+
+   TEST_ASSERT_EQUAL_INT(32, cell_value(output));
+   cell_value_set(input, 3);
+   TEST_ASSERT_EQUAL_INT(96, cell_value(output));
+
+   free_reactor(r);
+}
+
+struct cbinfo {
+   int last_value;
+   int times_called;
+};
+
+static void cb_record_last_value(void *obj, int v)
```

Comment:
How about `cb_spy` or something similar? I think that `spy` more succinctly communicates what this does and is an idiomatic term for this type of test double.